Updates variables.md with secrets hierarchy info

matthops opened the pull request:
# Why
I moved to add environment related variables to my eas.json file, but the app still built with my secrets on the expo.dev site, I was aware that this might occur so I caught it, but I think it's important to communicate the hierarchy of these methods to avoid confusion in the future when individuals expect to see different environment variables being used and don't understand why.

# How
Just added a line of text to clarify that secrets set on the website will override secrets set in eas.json or app.config.js files.

# Test Plan
I built my project with env variables in my EAS.json file and left the existing stored env variables on the website. I then checked the "spin up build environment" logs and observed it was using the stored variables. I then deleted the stored variables and rebuilt the project, checked the same logs and they were now using the values from my eas.json file.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

matthops committed:
Updates variables.md with secrets hierarchy info

I moved to add environment related variables to my eas.json file, but the app still built with my secrets on the expo.dev site, I was aware that this might occur so I caught it, but I think it's important to communicate the hierarchy of these methods to avoid confusion in the future when individuals expect to see different environment variables being used and don't understand why.

brentvatne commented on docs/pages/build-reference/variables.md:
```diff
@@ -241,10 +241,12 @@ The following are two possible alternative approaches, each with different trade
 
 <div style={{marginTop: -20, display: 'block'}} />
 
+> ⚠️  Secrets set on the Expo website or through the EAS CLI will override any secrets set within your eas.json or app.config.js files. 
```
you don't set any secrets in eas.json or app.config.js. those are unencrypted env vars

matthops replied:
![https://media.giphy.com/media/ggu8E0UeH7bINjvacc/giphy-downsized.gif](https://media.giphy.com/media/ggu8E0UeH7bINjvacc/giphy-downsized.gif)

matthops replied:
just updated, let me know if that works or if you want to see something different.

matthops committed:
Update variables.md with more accurate language

change "secret" to "env variables" in warning about overriding env variables in eas.json && app.config.js

brentvatne commented on docs/pages/build-reference/variables.md:
```diff
@@ -241,10 +241,12 @@ The following are two possible alternative approaches, each with different trade
 
 <div style={{marginTop: -20, display: 'block'}} />
 
+> ⚠️  Secrets set on the Expo website or through the EAS CLI will override any env variables with the same name set within your eas.json or app.config.js files. 
```
```suggestion
> ⚠️  Secrets set on the Expo website or through the EAS CLI will override any environment variables with the same name set within **eas.json**. 
```

not sure what the original intent was for including app.config.js, you should only set env vars in eas.json. you can read them in app.config.js if you want.

brentvatne committed:
Update docs/pages/build-reference/variables.md

brentvatne commented on docs/pages/build-reference/variables.md:
```diff
@@ -241,10 +241,14 @@ The following are two possible alternative approaches, each with different trade
 
+### How are naming collisions between secrets and the `env` field in eas.json handled?
+
+A secret created on the Expo website or with `eas secret:create` will take precedence over an environment variables of the same name that is set through the `env` field in **eas.json**. For example, if you create a secret with name `MY_TOKEN` and value `secret` and also set `"env": { "MY_TOKEN": "public" }` in your **eas.json**, then `process.env.MY_TOKEN` on EAS Build will evaluate to `secret`.
```
```suggestion
A secret created on the Expo website or with `eas secret:create` will take precedence over an environment variable of the same name that is set through the `env` field in **eas.json**. For example, if you create a secret with name `MY_TOKEN` and value `secret` and also set `"env": { "MY_TOKEN": "public" }` in your **eas.json**, then `process.env.MY_TOKEN` on EAS Build will evaluate to `secret`.
```

brentvatne committed:
Update docs/pages/build-reference/variables.md

brentvatne commented:
thanks for following up on this @matthops!
